stdlib/logging/config.pyi

```diff
@@ -15,9 +33,65 @@ if sys.version_info >= (3, 7):
 else:
     _Path = StrPath
 
+_Tkey = TypeVar("_Tkey", bound=Hashable)
+_TValue = TypeVar("_TValue")
+_T = TypeVar("_T")
+_V = TypeVar("_V")
+
 DEFAULT_LOGGING_CONFIG_PORT: int
 RESET_ERROR: int  # undocumented
 IDENTIFIER: Pattern[str]  # undocumented
+_listener: Optional[Any]
+dictConfigClass = DictConfigurator
+
+def _resolve(name: str) -> _V: ...
+def _strip_spaces(alist: List[str]) -> Iterable[_Tkey]: ...
+def _create_formatters(cp: Dict[_Tkey, Dict[_Tkey, _TValue]]) -> Dict[_Tkey, _TValue]: ...
+def _install_handlers(
+    cp: Dict[_Tkey, Dict[_Tkey, _TValue]], formatters: Dict[_Tkey, Dict[_Tkey, _TValue]]
+) -> Dict[_Tkey, _TValue]: ...
+def _handle_existing_loggers(
+    existing: List[Union[str, Logger]], child_loggers: List[Union[str, Logger]], disable_existing: bool
+) -> None: ...
+def _install_loggers(cp: Dict[_Tkey, Dict[_Tkey, _TValue]], handlers: Dict[_Tkey, _TValue], disable_existing: bool) -> None: ...
+def _clearExistingHandlers() -> None: ...
+
+class ConvertingMixin(object):
+    def convert_with_key(self, key: _Tkey, value: _TValue, replace: bool) -> Dict[_Tkey, _TValue]: ...
+    def convert(self, value: _TValue) -> Dict[_Tkey, _TValue]: ...
+
+class ConvertingDict(Generic[_Tkey, _TValue], ConvertingMixin):
+    def __getitem__(self, key: _Tkey) -> _TValue: ...
+    def get(self, key: _Tkey, default: Optional[_Tkey] = ...) -> _TValue: ...
+    def pop(self, key: _Tkey, default: Optional[_Tkey] = ...) -> _TValue: ...
+
+class ConvertingList(Generic[_T], ConvertingMixin):
+    @overload
+    def __getitem__(self, key: int) -> _T: ...
+    @overload
+    def __getitem__(self, key: slice) -> List[_T]: ...
+    def pop(self, idx: int) -> _TValue: ...
+
+class ConvertingTuple(Generic[_T], ConvertingMixin):
+    @overload
+    def __getitem__(self, key: int) -> _V: ...
```

Comment:
This seems to be a misunderstanding of how `TypeVar` is supposed to work.

When you have a class that is `Generic[_T]`, then inside the class, `_T` will be some type. But `__getitem__` uses `_V` and not `_T`, so `_V` should be determined from the arguments given to `__getitem__`. This is not possible, because `_V` is never mentioned in the arguments.

For a simpler example, consider these two functions:

```
# Meaningful: do_nothing(123) returns int, because `_T` will be the type of `123`
def do_nothing(arg: _T) -> _T: ...

# Does not make sense and causes pytype errors.
# What type should do_weird_thing(123) return?
def do_weird_thing(arg: int) -> _T: ...
```

Reply:
I got it.